### PR TITLE
SEP6-24 Remove movie review

### PR DESCRIPTION
**Pull request by RealFoXsTeR**
## How Has This Been Tested?

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

## Screenshots (if appropriate):

image
😄

### CODE REVIEW

File: src/MovieManagement/FunctionDtos/ReviewsResponseDto.cs [hunk 1]
```diff
@@ -4,5 +4,5 @@ public class ReviewsResponseDto
 {
     public int TotalPages { get; set; }
     public int Page { get; set; }
-    public IEnumerable<ReviewResponseDto> ReviewReponseDtos { get; set; } = default!;
+    public IEnumerable<ReviewResponseDto> MovieRatingDtos { get; set; } = default!;
```
Comment (stefaniatomuta) [inline src/MovieManagement/FunctionDtos/ReviewsResponseDto.cs:7]: Why do you have IEnumerable instead of IList?

File: src/MovieManagement.Functions/Dtos/GetRatingDto.cs [hunk 2]
```diff
@@ -2,6 +2,6 @@
 
 public class GetRatingDto
 {
-    public int? MovieId { get; set; }
-    public Guid? UserId { get; set; }
```
Comment (HolaSombreros) [inline src/MovieManagement.Functions/Dtos/GetRatingDto.cs:6]: Why is this changed? The fluent validation for id won't work otherwise

File: src/MovieManagement/appsettings.json [hunk 3]
```diff
@@ -30,7 +30,8 @@
     "GetMovieRatingsByIdsPath": "/GetMovieRatingsByIds",
     "QueryBuilder": "?",
     "PagePath": "page=",
-    "AndQueryBuilder": "&"
+    "AndQueryBuilder": "&",
+    "DeleteReviewPath": "/DeleteRating"
```
Comment (stefaniatomuta) [inline src/MovieManagement/appsettings.json:34]: What is the purpose of this? Here are the TMDB API paths

File: src/MovieManagement.Functions/Dtos/MovieRatingDto.cs [hunk 4]
```diff
@@ -2,8 +2,9 @@
...
 public class MovieRatingDto
 {
+    public Guid Id { get; set; }
     public int Rating { get; set; }
-    public string? Review { get; set; }
-    public DateTime? CreatedDate { get; set; }
-    public string? CreatedBy { get; set; }
+    public string Review { get; set; }
+    public DateTime CreatedDate { get; set; }
+    public UserDto CreatedBy { get; set; }
```
Comment (HolaSombreros) [inline src/MovieManagement.Functions/Dtos/MovieRatingDto.cs:9]: what is the reason for this change?

File: src/MovieManagement.Functions/Services/RatingService.cs [hunk 5]
```diff
@@ -71,13 +71,18 @@ public async Task<RatingResultDto> GetMovieRatings(GetRatingDto ratingDto, int p
...
-                users.FirstOrDefault(u => u.UserId == rating.UserId) 
+                users.First(u => u.UserId == rating.UserId) 
             select new MovieRatingDto
             {
+                Id = rating.RatingId,
                 Rating = (int)rating.Rating, 
                 Review = rating.Review, 
                 CreatedDate = rating.DateTime, 
-                CreatedBy = user?.Username
+                CreatedBy = new UserDto
```
Comment (HolaSombreros) [inline src/MovieManagement.Functions/Services/RatingService.cs:81]: Why do you need to return the user id as well?

File: src/MovieManagement.Functions/Dtos/MovieDto.cs [hunk 6]
```diff
@@ -3,7 +3,7 @@
 public class MovieDto
 {
     public int MovieId { get; set; }
-    public string? Title { get; set; }
-    public string? PosterUrl { get; set; }
-    public string? ReleaseDate { get; set; }
```
Comment (stefaniatomuta) [inline src/MovieManagement.Functions/Dtos/MovieDto.cs:8]: FluentValidation does not work if you remove the ? from strings, it will not allow the properties to be checked later in the validator, it  will throw an error. Better ask before making such changes, especially if you do not know the libarary

**Review by stefaniatomuta — CHANGES_REQUESTED** [review]

File: src/MovieManagement.Functions/Dtos/GetRatingDto.cs (hunk 2 above)
Comment (RealFoXsTeRDOTA) [reply src/MovieManagement.Functions/Dtos/GetRatingDto.cs:6]: Because I need access to the user id of the author of a review in order to determine whether I can remove the review or not.

File: src/MovieManagement.Functions/Dtos/MovieRatingDto.cs (hunk 4 above)
Comment (RealFoXsTeRDOTA) [reply src/MovieManagement.Functions/Dtos/MovieRatingDto.cs:9]: Because I need access to the user id of the author of a review in order to determine whether I can remove the review or not.
